docs/proposals/kubeedgeadm-scope.md

```diff
@@ -0,0 +1,122 @@
+
+## Scope of commands
+
+**Cloud commands:**
+
+`kubeedge cloud init`
+  - What is it?
+     * This command will be responsible to bring up KubeEdge cloud components like edge-controller and K8S (using kubeadm)
+   
+  - What shall be its scope ?
+    1. Check version of OS and install subsequently the required pre-requisites using supported steps. Currently we will support **ONLY** (Ubuntu & CentOS)
+    2. Check and install all the pre-requisites before executing edge-controller, which are
+        * docker (currently 18.06.0ce3-0~ubuntu) and check if service is up
+        * kubelet, kubeadm & kubectl (latest version)
+        * openssl (latest available in OS repos)
+    3. Generate certificates using openssl and save the certs in a predefined static path.
+    It will also compress the folder and display on the terminal so that user can pick it up and transfer it to edge node (VM/host) manually.
+    4. It will update the certificate information in `controller.yaml`
+    5. Start `kubeadm init`.
+
+       **NOTE:** If any issues or error reported from `kubeadm init`, `kubeedge cloud init` command shall be not responsible, as it may occur due to the environment. User has to resolve it. We are taking up this activity to have 1 click install approach for KubeEdge and not K8S.
+
+    6. Update `/etc/kubernetes/manifests/kube-apiserver.yaml` with below information
+    ```
+    - --insecure-port=8080
+    - --insecure-bind-address=0.0.0.0
+    ```
+
+    7. start edge-controller
+
+`kubeedge cloud reset`
+  - What is it? 
+    * This command will be responsible to bring down KubeEdge cloud components edge-controller and call `kubeadm reset` (to stop K8S)
+
+  - What shall be its scope ?
+    1. It shall get the nodes and execute `kubectl drain --force`.
+    2. Kill `edge-controller` process
+    3. Execute `kubeadm reset`
+
+**Edge commands:**
+
+`kubeedge node init`
+  - What is it? 
+    * This command will be responsible to install pre-requisites KubeEdge edge component (edge_core)
+
+  - What shall be its scope ?
+
+    1. Check version of OS and install subsequently the required pre-requisites using supported steps. Currently we will support **ONLY** (Ubuntu & CentOS)
+    2. Check and install all the pre-requisites before executing edge-controller, which are
+        * Docker (currently 18.06.0ce3-0~ubuntu) and check is service is up.
+        * mosquitto (latest available in OS repos) and check if running.
+        * kubelet
+    3. This command will take `--certPath` (string type) as mandatory option which shall be the certificates path; wherein the certs were transfered from cloud node and uncompressed. It will modify `$GOPATH/src/github.com/kubeedge/kubeedge/edge/conf/edge.yaml` file against `edgehub.websocket.certfile` and `edgehub.websocket.keyfile` fields.
```

Comment:
I suggest you first give the cli command with such flags before describing the steps 1,2,3...etc.

example from kubeadm : 

```sh
Usage:
  kubeadm join [flags]

Flags:
      --apiserver-advertise-address string            If the node should host a new control plane instance, the IP address the API Server will advertise it's listening on.
      --apiserver-bind-port int32                     If the node should host a new control plane instance, the port for the API Server to bind to. (default 6443)
      --config string                                 Path to kubeadm config file.
      --cri-socket string                             Specify the CRI socket to connect to. (default "/var/run/dockershim.sock")
      --discovery-file string                         A file or url from which to load cluster information.
      --discovery-token string                        A token used to validate cluster information fetched from the api server.
      --discovery-token-ca-cert-hash strings          For token-based discovery, validate that the root CA public key matches this hash (format: "<type>:<value>").
      --discovery-token-unsafe-skip-ca-verification   For token-based discovery, allow joining without --discovery-token-ca-cert-hash pinning.
      --experimental-control-plane                    Create a new control plane instance on this node
      --feature-gates string                          A set of key=value pairs that describe feature gates for various features. Options are:
                                                      Auditing=true|false (ALPHA - default=false)
                                                      CoreDNS=true|false (default=true)
                                                      DynamicKubeletConfig=true|false (BETA - default=false)
  -h, --help                                          help for join
```

Reply:
let's try to give the cli command help for every command in this design , which will help the reviewer to easily understand the usage.

Reply:
+1

Reply:
Have added the basic, required one's. Please check the updated document.